test/fixtures/rules/std_rule_cases/L059.yml

```
@@ -47,3 +79,197 @@ test_pass_special_symbols:
     SELECT 123 AS "SELECT"
+
+test_pass_column_reference_backtick:
+  pass_str: |
+    SELECT 123 AS foo;
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_fail_column_reference_backtick:
+  fail_str: |
+    SELECT 123 AS "foo";
+  pass_str: |
+    SELECT 123 AS `foo`;
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_table_reference_backtick:
+  pass_str: |
+    SELECT foo
+    FROM bar;
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_fail_table_reference_backtick:
+  fail_str: |
+    SELECT foo
+    FROM "bar";
+  fix_str: |
+    SELECT foo
+    FROM `bar`;
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_multiple_references_backtick:
+  pass_str: |
+    SELECT foo
+    FROM bar.baz;
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_fail_multiple_references_backtick:
+  fail_str: |
+    SELECT foo
+    FROM "bar"."baz";
+  fix_str: |
+    SELECT foo
+    FROM `bar`.`baz`;
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_whitespace_backtick:
+  pass_str: |
+    SELECT 123 AS `I cannot be unquoted`
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_special_symbols_backtick:
+  pass_str: |
+    SELECT 123 AS `I-c@nn0t-be~un-quoted`
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_reserved_keyword_backtick:
+  pass_str: |
+    SELECT 123 AS `SELECT`
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_ignored_as_contains_backtick:
+  pass_str: |
+    SELECT 123 AS "Ignored`as`contains`backticks"
+  configs:
+    rules:
+      L059:
+        preferred_quote_identifier: "`"
+
+test_pass_column_reference_backtick_forced:
+  pass_str: |
+    SELECT 123 AS `foo`;
+  configs:
+    rules:
+      L059:
+        force_quote_identifier: True
+        preferred_quote_identifier: "`"
+
+test_fail_column_reference_backtick_forced:
+  fail_str: |
+    SELECT 123 AS foo;
+  pass_str: |
+    SELECT 123 AS `foo`;
+  configs:
+    rules:
+      L059:
+        force_quote_identifier: True
+        preferred_quote_identifier: "`"
+
+test_pass_table_reference_backtick_forced:
+  pass_str: |
+    SELECT `foo`
+    FROM `bar`;
+  configs:
+    rules:
+      L059:
+        force_quote_identifier: True
+        preferred_quote_identifier: "`"
+
+test_fail_table_reference_backtick_forced:
+  fail_str: |
+    SELECT `foo`
+    FROM bar;
+  fix_str: |
+    SELECT `foo`
+    FROM `bar`;
+  configs:
+    rules:
+      L059:
+        force_quote_identifier: True
+        preferred_quote_identifier: "`"
+
+test_pass_multiple_references_backtick_forced:
+  pass_str: |
+    SELECT `foo`
+    FROM `bar`.`baz`;
+  configs:
+    rules:
+      L059:
+        force_quote_identifier: True
+        preferred_quote_identifier: "`"
+
```

It's very common in BigQuery to back tick the quote table so we should add that test case in:

```suggestion
test_pass_multiple_references_backtick_forced:
  pass_str: |
    SELECT `foo`
    FROM `bar`.`baz`;
  configs:
    rules:
      L059:
        force_quote_identifier: True
        preferred_quote_identifier: "`"

test_pass_multiple_references_backtick_forced:
  pass_str: |
    SELECT `foo`
    FROM `bar.baz`;
  configs:
    core:
      dialect: bigquery
    rules:
      L059:
        force_quote_identifier: True
        preferred_quote_identifier: "`"

```

---

Done in : https://github.com/niconoe-/sqlfluff/commit/4f912bd15120e18b1f2d4c1f33c8cefe811b6f27